examples/routing_service/routing_service_file_adapter/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,74 @@
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+set (CMAKE_CXX_STANDARD 11)
```

Comment:
Please, use the [CXX_STANDARD](https://cmake.org/cmake/help/v3.1/prop_tgt/CXX_STANDARD.html) property instead.

Reply:
You mean set it per target? Although in this case we just have "fileadapter".

What do you think about adding set(CMAKE_CXX_STANDARD_REQUIRED ON) as well?

Reply:
Also I am interested to know if there is any specific reason for not using CMAKE_CXX_STANDARD variable.

Reply:
I don't see the point of setting `CMAKE_CXX_STANDARD_REQUIRED` to `ON`.

The reason to not use `CMAKE_CXX_STANDARD` is simple: you are forcing the full project to use that CXX STANDARD. If this project is added as part of a bigger project (as we did for the ConnextDDS examples), you will be forcing the parent project. It is not a good idea to set things globally if they can be set per target.

Reply:
Got it. Thanks for the clarification Israel.

I used the CXX_STANDARD and CXX_STANDARD_REQUIRED properties. The reason for using the latter is we don't want the decay behavior for the compiler flags pertaining to C++11. This should make the CMake generate step fail instead of the compilation.